builder/builder_test.go

```diff
@@ -36,6 +38,73 @@ func (*queryAll) String() string {
 	return "all"
 }
 
+//nolint:gocritic // tooManyResultsChecker: this function needs to return multiple values for test setup
+func setupTestEnvironment(t *testing.T) (*testapp.App, *genesis.Genesis, *mempool.Pool, *localdb.DB, txstore.TxStore, state.Database, *bfttypes.EventBus) {
```

Comment:
It may be better to just create a `testEnvironment` struct instead of needing to suppress the lint warning.

Also, can we use the test environment in the `TestRollback` test?